compiler/rustc_data_structures/src/profiling.rs

```diff
@@ -400,6 +400,7 @@ impl SelfProfiler {
     ) -> Result<SelfProfiler, Box<dyn Error + Send + Sync>> {
         fs::create_dir_all(output_directory)?;
 
+        debug!(?crate_name);
```

Comment:
We probably don't want this debug statement, or at least, we don't want it in the current form 🙂

Reply:
I'm not sure what the policy on debug logging is - I'm happy to revert this, but I'm not sure what a better form would be.

Reply:
I think in general we try to include some kind of context about what is going on when this value is logged. Seeing for example 

```
[DEBUG]    rustc_data_structures::profiling     my_crate_name` 
```

isn't super useful but something like `debug!("Creating self-profiler: output_directory = {:?}, crate_name = {:?}, event_filters = {:?}", output_directory, crate_name, event_filters);` would add more context about what is being logged and could be useful for someone else in the future.

Reply:
Btw, what does the prefixed `?` do here?

Reply:
@camelid https://docs.rs/tracing/0.1.22/tracing/?search=#recording-fields